tests/fixtures/2500_faulty_samplesheet.csv

```diff
@@ -1,15 +1,15 @@
-FCID,Lane,SampleID,SampleRef,Index,Description,Control,Recipe,Operator,SampleProject
-HB07NADXX,1,SIB911A2_sureselect5,hg19,ACAGTG,959191,N,R1,NN,959191
-HB07NADXX,1,SIB910A3_sureselect6,hg19,GCCAAT,454557,N,R1,NN,454557
-HB07NADXX,1,SIB914A2_sureselect2,hg19,CGATGT,504910,N,R1,NN,504910
-HB07NADXX,1,SIB914A11_sureselect11,hg19,GGCTAC,504910,N,R1,NN,504910
-HB07NADXX,1,SIB914A12_sureselect12,hg19,CTTGTA,504910,N,R1,NN,504910
-HB07NADXX,1,SIB914A15_sureselect15,hg19,GAAACC,504910,N,R1,NN,504910
-HB07NADXX,1,SIB911A1_sureselect5,hg19,ACAGTG,959191,N,R1,NN,959191
-HB07NADXX,2,SIB911A1_sureselect4,hg19,TGACCA,959191,N,R1,NN,959191
-HB07NADXX,2,SIB911A2_sureselect5,hg19,ACAGTG,959191,N,R1,NN,959191
-HB07NADXX,2,SIB910A3_sureselect6,hg19,GCCAAT,454557,N,R1,NN,454557
-HB07NADXX,2,SIB914A2_sureselect2,hg19,CGATGT,504910,N,R1,NN,504910
-HB07NADXX,2,SIB914A11_sureselect11,hg19,GGCTAC,504910,N,R1,NN,504910
-HB07NADXX,2,SIB914A12_sureselect12,hg19,CTTGTA,504910,N,R1,NN,504910
-HB07NADXX,2,SIB914A15_sureselect15,hg19,GAAACC,504910,N,R1,NN,504910
+FCID,Lane,SampleID,SampleRef,index,index2,SampleName,Control,Recipe,Operator,Project
+HB07NADXX,1,SIB911A2_sureselect5,hg19,ACAGTG,,959191,N,R1,NN,959191
+HB07NADXX,1,SIB910A3_sureselect6,hg19,GCCAAT,,454557,N,R1,NN,454557
+HB07NADXX,1,SIB914A2_sureselect2,hg19,CGATGT,,504910,N,R1,NN,504910
+HB07NADXX,1,SIB914A11_sureselect11,hg19,GGCTAC,,504910,N,R1,NN,504910
+HB07NADXX,1,SIB914A12_sureselect12,hg19,CTTGTA,,504910,N,R1,NN,504910
+HB07NADXX,1,SIB914A15_sureselect15,hg19,GAAACC,,504910,N,R1,NN,504910
+HB07NADXX,1,SIB911A1_sureselect5,hg19,ACAGTG,,959191,N,R1,NN,959191
+HB07NADXX,2,SIB911A1_sureselect4,hg19,TGACCA,,959191,N,R1,NN,959191
+HB07NADXX,2,SIB911A2_sureselect5,hg19,ACAGTG,,959191,N,R1,NN,959191
+HB07NADXX,2,SIB910A3_sureselect6,hg19,GCCAAT,,454557,N,R1,NN,454557
+HB07NADXX,2,SIB914A2_sureselect2,hg19,CGATGT,,504910,N,R1,NN,504910
+HB07NADXX,2,SIB914A11_sureselect11,hg19,GGCTAC,,504910,N,R1,NN,504910
+HB07NADXX,2,SIB914A12_sureselect12,hg19,CTTGTA,,504910,N,R1,NN,504910
+HB07NADXX,2,SIB914A15_sureselect15,hg19,GAAACC,,504910,N,R1,NN,504910
```

Comment:
Is it just me or is this sample sheet and the tests/fixtures/2500_samplesheet.csv identical? Is any of them actually faulty? I can only see that the order of samples are different.

Reply:
I need to look into this, I just changed the header.

Reply:
Ran the unit test with the correct (non-faulty) sample sheet:
![image](https://user-images.githubusercontent.com/26384108/109964080-80818f80-7ced-11eb-8546-24b30eec31bf.png)

Unit test fails, exception is not raised. So whatever is faulty with the 2500_faulty_samplesheet.csv, it's working as it should!

Not investigating any further...